features/netsocket/nsapi_dns.h

```diff
@@ -117,6 +130,20 @@ nsapi_error_t nsapi_dns_query(S *stack, const char *host,
 nsapi_size_or_error_t nsapi_dns_query_multiple(NetworkStack *stack, const char *host,
         SocketAddress *addr, nsapi_size_t addr_count, nsapi_version_t version = NSAPI_IPv4);
 
+/** Query a domain name server for an IP address of a given hostname (asynchronous)
+ *
+ *  @param stack      Network stack as target for DNS query
+ *  @param host       Hostname to resolve
+ *  @param callback   Callback that is called for result
+ *  @param data       Caller defined data returned in callback
+ *  @param addr_count Number of addresses allocated in the array
+ *  @param version    IP version to resolve (defaults to NSAPI_IPv4)
+ *  @return           0 on success, negative error code on failure
+ *                    NSAPI_ERROR_DNS_FAILURE indicates the host could not be found
+ */
+nsapi_size_or_error_t nsapi_dns_query_multiple_async(NetworkStack *stack, const char *host,
+        NetworkStack::hostbyname_cb_t callback, void *data, nsapi_size_t addr_count, nsapi_version_t version = NSAPI_IPv4);
```

Comment:
Should there be a C verions of `nsapi_dns_query_multiple_async`? If so it'd need the extern "C" and stuff.